From 8e57177c71e9942f9ed2bb459247eaf134b9910a Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Tue, 2 Jul 2013 02:02:04 +0200
Subject: [PATCH 31/65] acerhk: drop optional /proc/driver/acerhk/info file

Simplify procfs usage in preparation of the transition scheduled for
kernel v3.10. Besides of the number of pending events queried via
get_nr_events(), all information provided by /proc/driver/acerhk/info
is dumped to dmesg as well.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 drivers/platform/x86/acerhk.c | 72 -------------------------------------------
 1 file changed, 72 deletions(-)

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -1248,60 +1248,6 @@ static int acerhk_proc_init(void)
 		}						\
 	} while (0)
 
-static int pc_proc_infos(char *buffer, int *len,
-			 off_t *begin, off_t offset, int size)
-{
-	PRINT_PROC("Acer hotkeys\n");
-	PRINT_PROC("Model(Type)\t: %d(", acerhk_series);
-	PRINT_PROC("Dritek)\n");
-
-	if (bios_routine != 0) {
-		PRINT_PROC("request handler\t: 0x%x\n", bios_routine);
-		if (cmos_index) {
-			PRINT_PROC("CMOS index\t: 0x%x\n", cmos_index);
-			PRINT_PROC("events pending\t: %u\n", get_nr_events());
-		} else {
-			PRINT_PROC("CMOS index\t: not available\n");
-		}
-
-		if (acerhk_polling_state == 1)
-			PRINT_PROC("kernel polling\t: active\n");
-		else
-			PRINT_PROC("kernel polling\t: inactive\n");
-
-		PRINT_PROC("autoswitch wlan\t: ");
-		if (autowlan == 1)
-			PRINT_PROC("enabled\n");
-		else
-			PRINT_PROC("disabled\n");
-	} else {
-		PRINT_PROC("request handler\t: not found\n");
-		PRINT_PROC("kernel polling\t: not possible\n");
-	}
-
-	/* model specific infos */
-	PRINT_PROC("use of Dritek EC: ");
-	PRINT_PROC("enabled\n");
-
-	return 1;
-}
-
-static int acerhk_proc_info(char *buffer, char **start, off_t offset,
-			    int size, int *eof, void *data)
-{
-	int len = 0;
-	off_t begin = 0;
-
-	*eof = pc_proc_infos(buffer, &len, &begin, offset, size);
-
-	if (offset >= begin + len)
-		return 0;
-
-	*start = buffer + (offset - begin);
-
-	return (size < begin + len - offset ? size : begin + len - offset);
-}
-
 static int acerhk_proc_key(char *buffer, char **start, off_t offset,
 			   int size, int *eof, void *data)
 {
@@ -1456,23 +1402,11 @@ static int acerhk_proc_init(void)
 		return 0;
 	}
 
-	/* now create several files, first general info ... */
-	entry = create_proc_read_entry("info", 0444, proc_acer_dir,
-				       acerhk_proc_info, NULL);
-
-	if (entry == NULL) {
-		pr_info("cannot create info file\n");
-		remove_proc_entry("driver/acerhk", NULL);
-
-		return 0;
-	}
-
 	/* ... last pressed key ... */
 	entry = create_proc_read_entry("key", 0444, proc_acer_dir,
 				       acerhk_proc_key, NULL);
 	if (entry == NULL) {
 		pr_info("cannot create key file\n");
-		remove_proc_entry("info", proc_acer_dir);
 		remove_proc_entry("driver/acerhk", NULL);
 
 		return 0;
@@ -1482,7 +1416,6 @@ static int acerhk_proc_init(void)
 	entry = create_proc_entry("led", 0220, proc_acer_dir);
 	if (entry == NULL) {
 		pr_info("cannot create LED file\n");
-		remove_proc_entry("info", proc_acer_dir);
 		remove_proc_entry("key", proc_acer_dir);
 		remove_proc_entry("driver/acerhk", NULL);
 		return 0;
@@ -1493,8 +1426,6 @@ static int acerhk_proc_init(void)
 	entry = create_proc_entry("wirelessled", 0220, proc_acer_dir);
 	if (entry == NULL) {
 		pr_info("cannot create wirelessled file\n");
-		remove_proc_entry("info",
-				 proc_acer_dir);
 		remove_proc_entry("key",
 				 proc_acer_dir);
 		remove_proc_entry("led",
@@ -1509,7 +1440,6 @@ static int acerhk_proc_init(void)
 	entry = create_proc_entry("blueled", 0220, proc_acer_dir);
 	if (entry == NULL) {
 		pr_info("cannot create blueled file\n");
-		remove_proc_entry("info", proc_acer_dir);
 		remove_proc_entry("key", proc_acer_dir);
 		remove_proc_entry("led", proc_acer_dir);
 		remove_proc_entry("wirelessled", proc_acer_dir);
@@ -1525,8 +1455,6 @@ static int acerhk_proc_init(void)
 static void acerhk_proc_cleanup(void)
 {
 	if (proc_acer_dir) {
-		remove_proc_entry("info", proc_acer_dir);
-
 		/* On dritek type hardware key file is already removed */
 		remove_proc_entry("led", proc_acer_dir);
 		remove_proc_entry("wirelessled", proc_acer_dir);
